test/framework/integration_framework/fake_peer/fake_peer.cpp

```diff
@@ -282,5 +306,9 @@ namespace integration_framework {
                                                         request);
     }
 
+    void FakePeer::ensureInitialized() {
+      BOOST_VERIFY_MSG(initialized_, "Instance not initialized!");
```

Comment:
Why we print here error message instead of just calling `initialize()` ?

Reply:
I would prefer the users of this interface to see explicitly where they operate on a not yet initialized object, and when they initialize it.